var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -93,26 +93,19 @@ def url_for_version(self, version):
 
     # High-level API of HDF5 1.8.9 or later is required for netCDF-4 support:
     # http://www.unidata.ucar.edu/software/netcdf/docs/getting_and_building_netcdf.html
-    depends_on('hdf5@1.8.9:+hl~mpi', when='~mpi')
-    depends_on('hdf5@1.8.9:+hl+mpi', when='+mpi')
+    depends_on('hdf5@1.8.9:+hl')
```

Comment:
We should mention a non-default value for the variant of `hdf5` separately. Otherwise, we will need to mention it in all other constraints that we have.